app/scripts/controllers/table.js

```diff
@@ -62,5 +65,62 @@ app.controller('TableCtrl',['$location','$scope','$http','DatasetService', funct
     DatasetService.putProducts($scope.forams.slice($scope.currentSet.start,$scope.currentSet.stop+1));
     $location.path("/charts");
   };
-
+    $scope.generateChart = function(){
+      console.log('przenosimy do charts');
+      DatasetService.putProducts($scope.forams.slice($scope.currentSet.start,$scope.currentSet.stop+1));
+      $location.path("/charts");
+    };
+    $scope.hasFilters = false;
+    $scope.filters = [];
+    $scope.filterData = function() {
+      var i;
+      for(i in $scope.filters) {
+        $scope.filters[i].param.trim();
+      }
+      console.log($scope.filters);
+    };
+    $scope.addFilter = function() {
+      $scope.filters.push({});
+      $scope.hasFilters = true;
+    };
+    $scope.clearFilters = function() {
+      $scope.hasFilters = false;
+      $scope.filters = [];
+    };
+    $scope.deleteFilter = function(index) {
+      $scope.filters.splice(index,1);
+      if($scope.filters.length == 0) $scope.hasFilters = false;
+    };
+    $scope.maxForamsWithoutWarning = 100; // TODO move to configuration file
+    $scope.getForams = function() {
+        var numberOfForams = getNumberOfForamsInDb();
+        if(numberOfForams > 100) {
+          dialog = ngDialog.open({ template: 'popupTmpl.html', scope: $scope });
+        }
+    };
+    // Mock method to return number of forams in db
```

Comment:
```javascript
// TODO: Implement integration with backend to fetch number of forams in db
```
(jeżeli tylko takie komentarze będę umieszczał, to znaczy że bardzo dobry kod stworzyłeś :))